Generate project.json in %TEMP% rather than in install location

chuckries opened the pull request:
With this change we generate the project.json in %temp%\getclrdbg and
restore/publish from this directory into the final install directory. This
avoids having the project.json in the final install directory.

chuckries commented:
@gregg-miskelly @@NCarlsonMSFT @kelltrick

NCarlsonMSFT commented on scripts/GetClrDbg.ps1:
```diff
@@ -104,13 +104,14 @@ if (-not $RuntimeID) {
+$TempPath = Join-Path -Path $env:TEMP -ChildPath "getclrdbg"
+if (-not (Test-Path -Path $TempPath -PathType Container)) {
+    New-Item -ItemType Directory -Force -Path $TempPath
 }
```
Should you delete the folder if it exists? I'm a little concerned that the file gets cached somewhere I don't know about and may not realize the need to clean to get an update.

chuckries replied:
The way project.json is generated, it will be recreated each time this script is run, nothing is cached.

NCarlsonMSFT replied:
Got it, I see above it will override the contents regardless. Is there any other file that may be left behind that could be a problem? If not then no need to delete the folder.

chuckries replied:
I don't believe so. 

---
In reply to: [63795897](https://github.com/Microsoft/MIEngine/pull/332#discussion_r63795897) [](ancestors = 63795897)

rajkumar42 commented on scripts/GetClrDbg.ps1:
```diff
@@ -104,13 +104,14 @@ if (-not $RuntimeID) {
-if (-not (Test-Path -Path $InstallPath -PathType Container)) {
-    New-Item -ItemType Directory -Force -Path $InstallPath
+# create the temp folder if it does not exist
+$TempPath = Join-Path -Path $env:TEMP -ChildPath "getclrdbg"
```
I'd suggest using GetTempFileName()

$TempPath = [System.IO.Path]::GetTempFileName();

chuckries replied:
I don't think this is the right way to go, but you make a good point about two things trying to run this script simultaneously. 

---
In reply to: [63795430](https://github.com/Microsoft/MIEngine/pull/332#discussion_r63795430) [](ancestors = 63795430)

gregg-miskelly commented:
> dotnet publish -r $RuntimeID -o $InstallPath

'dotnet publish' will create the directory if it doesn't exist?

---
Refers to: scripts/GetClrDbg.ps1:126 in 77c55ee. [](commit_id = 77c55eeb98d345999bdf14a49e6daa25ddc029b9, deletion_comment = False)

chuckries commented:
@gregg-miskelly yes.

chuckries commented:
> dotnet publish -r $RuntimeID -o $InstallPath

yes.

---
In reply to: [220180314](https://github.com/Microsoft/MIEngine/pull/332#issuecomment-220180314) [](ancestors = 220180314)

---
Refers to: scripts/GetClrDbg.ps1:126 in 77c55ee. [](commit_id = 77c55eeb98d345999bdf14a49e6daa25ddc029b9, deletion_comment = False)

kelltrick commented on scripts/GetClrDbg.ps1:
```diff
@@ -104,13 +104,14 @@ if (-not $RuntimeID) {
-    New-Item -ItemType Directory -Force -Path $InstallPath
+# create the temp folder if it does not exist
+$TempPath = Join-Path -Path $env:TEMP -ChildPath "getclrdbg"
+if (-not (Test-Path -Path $TempPath -PathType Container)) {
```
There is no point to having ```-PathType Container``` if you don't deal with having a file named 'getclrdbg'

kelltrick commented:
lgtm

chuckries commented:
Updated this to created a directory named from a new Guid under %temp% and also fixed an issue with giving the script a relative path as an install location. This should fix any caching/simultaneous use issues.

gregg-miskelly commented on scripts/GetClrDbg.ps1:
```diff
@@ -104,13 +104,20 @@ if (-not $RuntimeID) {
-    New-Item -ItemType Directory -Force -Path $InstallPath
+# create the temp folder if it does not exist
+$GuidString = [System.Guid]::NewGuid()
+$TempPath = Join-Path -Path $env:TEMP -ChildPath $GuidString
```
If you are going to do it this way (which sounds reasonable to me) you should probably also delete the directory at the end of the script.

chuckries replied:
done. 

---
In reply to: [63798580](https://github.com/Microsoft/MIEngine/pull/332#discussion_r63798580) [](ancestors = 63798580)
